src/adr/0032-distributed-mail-queue-cleanup.md

```diff
@@ -0,0 +1,34 @@
+# 32. Distributed Mail Queue Cleanup
+
+Date: 2020-04-13
+
+## Status
+
+Proposed
+
+## Context
+
+Read [Distributed Mail Queue](0031-distributed-mail-queue.md) for full context.
+
+**enqueuedMailsV3** and **deletedMailsV2** is never cleanup and the corresponding blobs are always referenced. This is not
+ideal both from a privacy and space storage costs point of view.
+
+## Decision
+
+Add a new `contentStart` table referencing the point in time from which a given mailQueue holds data, for each mail queue.
+
+The values contained between `contentStart` and `browseStart` can safely be deleted.
+
+We can perform this cleanup upon `browseStartUpdate`: once finished we can browse then delete content of **enqueuedMailsV3**
+and **deletedMailsV2** contained between `contentStart` and the new `browseStart` then we can safely set `contentStart` 
+to the new `browseStart`.
```

Comment:
what about the grace period?

Reply:
grace period is applied upon browse start update.

content before browseStart can safely be considered deletable, and is applicatively no longer exposed. We don't need an additional grace period mechanism here. I will add an explanation.